src/formatDate.js

```diff
@@ -50,7 +50,59 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const divider = fromFormat[fromFormat.length - 1];
+  const toDivider = toFormat[toFormat.length - 1];
+  const splittedComingDate = date.split(divider);
+
+  const comingYearIndex = fromFormat.findIndex((value) => value[0] === 'Y');
+  const comingMonthIndex = fromFormat.findIndex((value) => value[0] === 'M');
+  const comingDayIndex = fromFormat.findIndex((value) => value[0] === 'D');
+
+  const comingYear = splittedComingDate[comingYearIndex];
+  const comingMonth = splittedComingDate[comingMonthIndex];
+  const comingDay = splittedComingDate[comingDayIndex];
+
+  let formattedDate = '';
+
+  for (const format of toFormat) {
+    switch (format[0]) {
+      case 'Y':
+        const comingYearLength = fromFormat[comingYearIndex].length;
+        const comingFormatIsDifferent = format.length !== comingYearLength;
+
+        if (comingFormatIsDifferent) {
+          if (format.length === 4) {
+            // we are here in case we have 2-digits year
+            // in `date` and `fromFormat
```

Comment:
Remove comment. Try to write obvious code with meaningful variable and function naming